upstream-sha256: 1df26c6a0beed7a81ab5b50660f92a17af52ecd0ac255b5764e6c20da50ca69d
actual-sha256: 75e863e8a7fd0bf0b431fa55fcdd81dc145c9c568fb54340616dde29af5a06a9
--- upstream.expected
+++ tolk.actual
@@ -18,7 +18,7 @@
   float val4 = (*(data4_16+(lidx1+4)));
   float val5 = (*(data4_16+(lidx1+8)));
   float4 val6 = (*((device float4*)((data3_16+alu0))));
-  *(data0_16+(lidx1+alu0)) = ((*(buf0+0))+(val6.x*val2)+(val6.y*val4)+(val6.z*val5)+(val6.w*val3));
+  *(data0_16+(lidx1+alu0)) = ((*(buf0+0))+(0.0f+(val6.x*val2)+(val6.y*val4)+(val6.z*val5)+(val6.w*val3)));
 }
 ---
 #include <metal_stdlib>
@@ -33,7 +33,7 @@
   threadgroup __attribute__((aligned(16))) float buf4[16];
   int lidx0 = lid.x; /* 16 */
   float val0 = (*(data1_16+lidx0));
-  *(buf4+lidx0) = (val0*val0);
+  *(buf4+lidx0) = (0.0f+(val0*val0));
   threadgroup_barrier(mem_flags::mem_threadgroup);
   *(buf0+0) = 0.0f;
   for (int Ridx106 = 0; Ridx106 < 16; Ridx106++) {
